src/main/java/command/Parser.java

```
@@ -0,0 +1,173 @@
+package command;
+
+import task.*;
+
+import java.util.Locale;
+
+import static java.lang.Integer.parseInt;
+
+public class Parser {
+    // Interprets user input
+    Ui ui;
+    TaskList list;
+    public Parser(Ui ui, TaskList list) {
+        this.ui = ui;
+        this.list = list;
+    }
+
+    public void processInput(String input) {
+        String[] inputAnalyzed = input.split(" ");
+        if (input.contains("#")) {
+            ui.showInvalidInputError("Illegal character '#'.");
+            return;
```

Maybe we can throw an `InvalidInputException` for this?